src/content/guides/packages.md

```diff
@@ -261,6 +261,174 @@ To update `pubspec.lock` run `dart pub get` without `--enforce-lockfile`.
+  Leading to confusion when context switching between the packages.
+* If you open the root folder in your IDE, the dart analyzer will have to have a
+  separate analysis contexts for each package.
+
+Pub allows you to organize your repository as a "workspace" using a single
+shared resolution for all your packages.
+
+To create a workspace:
+
+* Add a `pubspec.yaml` at the repository root directory with a `workspace` entry
+  enummerating the paths to the packages of the respository (the workspace
+  packages):
+
+```yaml
+name: _
+environment:
+  sdk: ^3.6.0
+workspace:
+  - packages/helper
+  - packages/client_package
+  - packages/server_package
+```
+
+* For each of the existing `pubspec.yaml`s, make sure their sdk constraint is at
+  least `^3.6.0` and add a `resolution` entry:
+
+  ```yaml
+  environment:
+    sdk: ^3.6.0
+  resolution: workspace
+  ```
+
+* Run `dart pub get` anywhere in the repository, and this will:
+  * Create a single `pubspec.lock` next to the root `pubspec.yaml` that contains
+    the a resolution of all the `dependencies` and `dev_dependencies` of all the
+    workspace packages. 
+  * Create a single shared `.dart_tool/package_config.json` that maps package
+    names to file locations.
+  * Delete any other existing `pubspec.lock` and
+    `.dart_tool/package_config.json` files next to workspace packages.
+
+Now the file structure looks like this:
+
+```console
+/
+  packages/
+    shared/
+      pubspec.yaml
+    client_package/
+      pubspec.yaml
+    server_package/
+      pubspec.yaml
+  pubspec.yaml
+  pubspec.lock
+  .dart_tool/package_config.json
+```
+
+:::version-note
+Pub workspaces was introduced in Dart 3.6.0.
+
+To use pub workspaces, all your workspace packages (but not your dependencies)
+must have a sdk version constraint of `^3.6.0` or higher.
+:::
+
+### Interdependencies between workspace packages
+
+If any of the workspace packages depend on each other, they will automatically
+resolve to the one in the workspace, regardless of the source.
+
+Eg. `packages/client_package/pubspec.yaml` might depend on `shared`:
+
+```yaml
+dependencies:
+  shared: 
+    hosted: https://pub.dev
+    version: ^2.3.0
+```
+
+When resolved inside the workspace, the _local_ version of `shared` will be
+used. 
+
+The local version of `shared` would still have to match the constraint
+(`^2.3.0`) though.
+
+But when the package is consumed as a dependency without being part of the
+workspace, the original source (here `hosted`) is used.
+
+So if `client_package` is published to pub.dev and someone depends on it, they
+will get the hosted version of `shared` as a transitive dependency.
+
+### Dependency overrides in a workspace
+
+All `dependency_overrides:` sections in the workspace packages are respected,
+and you can also place a `pubspec_overrides.yaml` file next to any of the
+workspace `pubspec.yaml` files.
+
+You can only override a package once in the workspace.
```

Comment:
```suggestion
You can only override a package once in the workspace. To keep overrides organized
it can be preferable to keep `dependency_overrides` in the root `pubspec.yaml`.
```